### PR TITLE
Add person, student, and teacher classes

### PR DESCRIPTION
**Pull request by DebyGrey**
In this project, I

- Created class Person with the following:
  -  Instance vars: @id, @name, and @age.
  - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
  - Getters for @id, @name, and @age.
  - Setters for @name and @age.
  - Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
  - Public method can_use_services? that returns true if person is of age or if they have permission from parents.
- Created class Student with the following:
  - Inherits from Person.
  - Constructor extends parent's constructor by adding @classroom and a parameter for it.
  - Method play_hooky that returns "¯\(ツ)/¯".
- Created class Teacher with the following:
   - Inherits from Person.
   - Constructor extends parent's constructor by adding @specialization and a parameter for it.
   - Override can_use_services? so it always returns true.
- Ensured that each class was saved in separate files.

### CODE REVIEW

**Comment by DebyGrey** [comment]
Screenshot 2023-08-04 at 17 28 30

From the screenshot, there are no linter errors in the console and I have pushed accordingly.

**Review by laiifuu — COMMENTED** [review]
## Hi @DebyGrey :wave: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project where linters check isn't passing. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  def initialize(age, name: 'unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id
```
Comment (christianbiring1) [inline person.rb]: - [x] Could you make sure your getters and setters are always coming before initializing your variables? i.e always make sure they are on top of the class for best practice before the `initialize method`

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @DebyGrey , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 


## Highlights
- [ ] Great work on implementing the changes as requested by the previous reviewer :+1: 
- [ ] Good job on adding the classes `Person`, `Student`, `Teacher` as required :100: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (DebyGrey) [reply person.rb]: @christianbiring1 Okay, will do that now. Thanks

**Review by emmiiorji — APPROVED** [review]
Hi @DebyGrey, 

# Status: Approved ✅

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<div align='center'>

</div>


## Highlights✔️

- Requested changes have been implemented

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**